### ISSUE TITLE
Cannot read property 'data' of null in replicate.js

### ISSUE THREAD

**Issue by pfrazee** [mafintosh/hypercore#146]
I got this report from somebody building from source (https://github.com/beakerbrowser/beaker/issues/919) so there's a good chance theyre using the latest hypercore. Here's the error:

```
Uncaught Exception:
TypeError: Cannot read property 'data' of null
    at Request.onvalue [as_callback] ($PATH_TO_BEAKER/app/node_modules/hypercore/lib/replicate.js:135:19)
    at Request.callback ($PATH_TO_BEAKER/app/node_modules/random-access-storage/index.js:150:8)
    at onread ($PATH_TO_BEAKER/app/node_modules/hyperdrive/node_modules/random-access-file/index.js:80:31)
    at FSReqWrap.wrapper [as oncomplete] (fs.js:658:17)
```

In https://github.com/mafintosh/hypercore/commit/e7b2b51d2836c6c1808432c78649da84c3b13016#diff-c169e1e72632697264bcd930cc444020 we set `.stream` to null on close, and I think it's possible for some async work to finish after that (eg https://github.com/mafintosh/hypercore/blob/e7b2b51d2836c6c1808432c78649da84c3b13016/lib/replicate.js#L151). I think the solution is to just return early in https://github.com/mafintosh/hypercore/blob/e7b2b51d2836c6c1808432c78649da84c3b13016/lib/replicate.js#L118 if `self.stream` is falsy.

**Comment by mafintosh** [comment]
You're right, nice catch. Fixed in 6.12.7

**Comment by pfrazee** [comment]
Thanks! I had a PR for this too, we can close it https://github.com/mafintosh/hypercore/pull/147
